daemons/lib-copy-offload/copy-offload.c

```diff
@@ -307,8 +307,25 @@ int copy_offload_copy(COPY_OFFLOAD *offload, char *compute_name, char *workflow_
     snprintf(urlbuf, COPY_OFFLOAD_URL_SIZE, "%s://%s/trial", offload->proto, *offload->host_and_port);
     curl_easy_setopt(offload->curl, CURLOPT_URL, urlbuf);
 
-    char *offload_req = "{\"computeName\": \"%s\", \"workflowName\": \"%s\", \"sourcePath\": \"%s\", \"destinationPath\": \"%s\", \"dmProfile\": \"copy-offload-nonmpi\", \"dryrun\": true, \"dcpOptions\": \"\", \"logStdout\": true, \"storeStdout\": false}";
-    snprintf(postbuf, COPY_OFFLOAD_POST_SIZE, offload_req, compute_name, workflow_name, source_path, dest_path);
+    if (profile_name == NULL) {
+        // We cannot use the default DM profile in k8s because it has setpriv commands to change
+        // from root to a non-root user. Those setpriv commands will fail in the copyoffload
+        // container since it is running as non-root. Therefore, we need to use the
+        // copy-offload-default profile and not the k8s default.
+        profile_name = "copy-offload-default";
+    }
+
+    char *offload_req =
+        "{\"computeName\": \"%s\", "
+        "\"workflowName\": \"%s\", "
+        "\"sourcePath\": \"%s\", "
+        "\"destinationPath\": \"%s\", "
+        "\"dmProfile\": \"%s\", "
+        "\"dryrun\": false, "
+        "\"dcpOptions\": \"\", "
+        "\"logStdout\": true, "
+        "\"storeStdout\": false}";
+    snprintf(postbuf, COPY_OFFLOAD_POST_SIZE, offload_req, compute_name, workflow_name, source_path, dest_path, profile_name);
```

Comment:
We should check the return value. Make sure we didn't run out of buffer space.

Reply:
Ok updated.